ocaml/xapi/xapi_pool.ml

```diff
@@ -3325,6 +3321,28 @@ let sync_updates ~__context ~self ~force =
       List.iter (fun x -> create_pool_repository ~__context ~self:x) enabled ;
       set_available_updates ~__context
 
+let check_update_readiness ~__context ~self ~requires_reboot =
```

Comment:
by this logic, we consider a pool 'ready to upgrade' if there are no miscreant VMs, and if all the hosts are up

what about if there's a cluster running, or HA is enabled, etc. ?

Reply:
Yeah, it just so happens that our update service already checks for HA and WLB, and disables them proactively. I would still add checks for them for extra safety though.

Reply:
I've added the HA check now. We are not currently convinced that anything else is needed.